docs/dev-environment/setting-up-dev-environment.md

```diff
@@ -54,73 +55,25 @@ For single build
-```
-nvm use 4.2
-npm start --production
-```
-in the database-and-ghost repository.
+Make sure to keep watching the build window, in case there is a syntax error or something else preventing builds. Your code will not recompile if there are build errors.
 
 # Run server
 
-You need to re-run the server to isomorphically serve the latest
-version. However, if you are just changing frontend, you can choose
-to not restart the server on change, and just refresh the page, and the frontend code will still
-be up to date time. The isomorphism will be broken, but it's not
-important for development.
+You need to re-run the server every time the code is rebuilt to serve the latest version. However, if you are just changing frontend, you can choose to not restart the server on change, and just refresh the page, and the frontend code will still be up to date time. The isomorphism will be broken, but it's not important for development.
 
 ```
-npm start # or node build/server.js
-```
-
-If you are not restarting the server you may have to disable caching for refresh to work. There's an option in Chrome
-dev tools to do this.
-
-# Convenience setup
-
-To manage your node versions it can be nice setting up some scripts, one way of doing that is using `virtualenv` and `virtualenvwrapper` that is normally used for Python development but can still be a nice tool. If you install these two packages a nice setup can be created by making two virtual environments as follows:
-
+npm start
 ```
-mkvirtualenv -a path/to/gazelle-server gazelle_server
-mkvirtualenv -a path/to/database-and-ghost-blog-server gazelle_ghost
-```
-
-Where the last argument is the names of the virtual environments.
-
-You can now setup some scripts to run when you `activate` these virtual environments. By adding the following to `~/.virtualenvs/gazelle_server/bin/postactivate`:
-
-```bash
-nvm use 9.3.0
-```
-
-and to `~/.virtualenvs/gazelle_server/bin/postactivate`:
-
-```bash
-nvm use 4.2
-npm start --production # This is just personal preference as I only ever visit this repo to start the ghost blog
-```
-
-Now when you write `workon gazelle_server` from any directory your current working directory will move to the `gazelle-server` repo and the correct node version will be started for you.
-
-When you write `workon gazelle_ghost` your current working directory will then move to the `database-and-ghost-blog-server` repo, the node version will be correctly changed and the server will start.
-
-> NOTE: `workon` also has autocomplete functionality which is a nice detail
 
-To exit any of these virtual environments at any point (though unless you're actually writing python the virtual env itself won't actually do anything, we're just using it as a nice way to write some postactivate scripts), just run `deactivate`.
+If you are not restarting the server you may have to disable caching for refresh to work. There's an option in Chrome dev tools to do this.
```

Comment:
This fact has given me errors that took important dev time away for debugging purposes. I think, ideally, we add in a flag `--caching true` whenever in dev mode, so that caching is off by default in dev mode.

Reply:
It's a browser setting I'm afraid, so not anything we can do about it like that. We'd have to add hot reloading or hot module reloading to get the desired effect, which is possible but a sizeable project

Reply:
Or actually what we can do is in dev mode we can recompute the hash for the client script, that I could add

Reply:
That is awesome, @emilgoldsmith good job! I do recommend though that this is a setting that can be turned on/off, to test how the catching works on client side. If you wanna do that now, great -- if not, let's push it in front of us and merge this in :)

Reply:
I don't think it's anything to test really, before it was just straight up wrong behaviour as it was sending a client script with an old hash without updating it, which was fine in dev, and in production we'd never update it without restarting the server, so I don't think there's much value in testing that wrong behaviour